--- Arduino/libraries/FabGL/src/fabutils.h	2022-08-30 00:11:28.000000000 +0900
+++ Arduino/FabGL/n80FabGL/patch1.0.9/src/fabutils.h	2022-10-02 11:58:46.679726034 +0900
@@ -884,7 +884,8 @@
    *     // Mount SD Card
    *     FileBrowser::mountSDCard(false, "/sdcard");
    */
-  static bool mountSDCard(bool formatOnFail, char const * mountPath, size_t maxFiles = 4, int allocationUnitSize = 16 * 1024, int MISO = 16, int MOSI = 17, int CLK = 14, int CS = 13);
+  //static bool mountSDCard(bool formatOnFail, char const * mountPath, size_t maxFiles = 4, int allocationUnitSize = 16 * 1024, int MISO = 16, int MOSI = 17, int CLK = 14, int CS = 13);
+  static bool mountSDCard(bool formatOnFail, char const * mountPath, size_t maxFiles = 4, int allocationUnitSize = 16 * 1024, int MISO = 12, int MOSI = 13, int CLK = 14, int CS = 15);
 
   /**
    * @brief Remounts SDCard filesystem, using the same parameters
--- Arduino/libraries/FabGL/src/fabutils.cpp	2022-08-30 00:11:28.000000000 +0900
+++ Arduino/FabGL/n80FabGL/patch1.0.9/src/fabutils.cpp	2022-10-02 11:57:02.215361859 +0900
@@ -1241,8 +1241,8 @@
       MOSI = 12;
       break;
     case ChipPackage::ESP32D0WDQ5:
-      MISO = 35;
-      MOSI = 12;
+      //MISO = 35;
+      //MOSI = 12;
       break;
     default:
       break;
--- Arduino/libraries/FabGL/src/inputbox.cpp	2022-08-30 00:11:28.000000000 +0900
+++ Arduino/FabGL/n80FabGL/patch1.0.9/src/inputbox.cpp	2022-10-02 11:59:40.063909869 +0900
@@ -90,7 +90,8 @@
 
   // setup keyboard and mouse
   if (!PS2Controller::initialized())
-    PS2Controller::begin(PS2Preset::KeyboardPort0_MousePort1, KbdMode::GenerateVirtualKeys);
+    //PS2Controller::begin(PS2Preset::KeyboardPort0_MousePort1, KbdMode::GenerateVirtualKeys);
+    PS2Controller::begin(PS2Preset::KeyboardPort0, KbdMode::GenerateVirtualKeys);
   else
     PS2Controller::keyboard()->enableVirtualKeys(true, true);
 }
--- Arduino/libraries/FabGL/src/dispdrivers/vgabasecontroller.cpp	2022-08-30 00:11:28.000000000 +0900
+++ Arduino/FabGL/n80FabGL/patch1.0.9/src/dispdrivers/vgabasecontroller.cpp	2022-10-02 11:56:07.183167518 +0900
@@ -117,7 +117,8 @@
 // initializer for default configuration
 void VGABaseController::begin()
 {
-  begin(GPIO_NUM_22, GPIO_NUM_21, GPIO_NUM_19, GPIO_NUM_18, GPIO_NUM_5, GPIO_NUM_4, GPIO_NUM_23, GPIO_NUM_15);
+  //begin(GPIO_NUM_22, GPIO_NUM_21, GPIO_NUM_19, GPIO_NUM_18, GPIO_NUM_5, GPIO_NUM_4, GPIO_NUM_23, GPIO_NUM_15);
+  begin(GPIO_NUM_22, GPIO_NUM_21, GPIO_NUM_19, GPIO_NUM_18, GPIO_NUM_5, GPIO_NUM_4, GPIO_NUM_23, GPIO_NUM_27);
 }
 
 
